newlib/libc/machine/riscv/memcpy.c

```diff
@@ -10,46 +11,101 @@
 */
 
 #if defined(PREFER_SIZE_OVER_SPEED) || defined(__OPTIMIZE_SIZE__)
-//memcpy defined in memcpy-asm.S
+// memcpy defined in memcpy-asm.S
 #else
 
-#include <string.h>
-#include <stdint.h>
 #include "../../string/local.h"
+#include <stdint.h>
+#include <string.h>
 
 #define unlikely(X) __builtin_expect (!!(X), 0)
 
+static inline long
+__load_word (const unsigned char *p)
```

Comment:
this is an xlen load.
if you had rv64 with abi ilp32, you would still want this to work on 64b data.
In general, you should get rid of all cases of long and sizeof(long).
Use https://github.com/puranikvinit/newlib/blob/main/newlib/libc/machine/riscv/xlenint.h
Instead of sizeof(long), use SZREG from sys/asm.h